app/models/mhv_account.rb

```diff
@@ -6,71 +6,74 @@ class MhvAccount < ActiveRecord::Base
+    state(*ALL_STATES)
 
+    # NOTE: This is eligibility for account creation or upgrade, not for access to services.
     event :check_eligibility do
-      transitions from: ALL_STATES, to: :existing, if: :preexisting_account?
-      transitions from: ALL_STATES, to: :ineligible, unless: :eligible?
-      transitions from: ALL_STATES, to: :upgraded, if: :previously_upgraded?
-      transitions from: ALL_STATES, to: :registered, if: :previously_registered?
-      transitions from: ALL_STATES, to: :unknown
+      transitions from: ALL_STATES, to: :needs_identity_verification, unless: :identity_proofed?
+      transitions from: ALL_STATES, to: :needs_ssn_resolution, if: :ssn_mismatch?
+      transitions from: ALL_STATES, to: :needs_va_patient, unless: :va_patient?
+      transitions from: ALL_STATES, to: :has_deactivated_mhv_ids, if: :deactivated_mhv_ids?
+      transitions from: ALL_STATES, to: :has_multiple_active_mhv_ids, if: :multiple_active_mhv_ids?
+      transitions from: ALL_STATES, to: :needs_terms_acceptance, unless: :terms_and_conditions_accepted?
+      transitions from: ALL_STATES, to: :eligible
     end
 
-    event :check_terms_acceptance do
-      transitions from: ALL_STATES - %i[existing ineligible],
-                  to: :needs_terms_acceptance, unless: :terms_and_conditions_accepted?
+    event :check_account_state do
+      transitions from: %i[eligible], to: :no_account, unless: :exists?
+      # The states below this line and the next comment will be removed when reintroducing upgrade.665
+      transitions from: %i[eligible], to: :upgraded, if: :previously_upgraded?
+      transitions from: %i[eligible], to: :registered, if: :previously_registered?
+      # this could mean that vets.gov created / upgraded before we started tracking mhv_ids
+      transitions from: %i[eligible], to: :existing
     end
 
     event :register do
-      transitions from: %i[unknown register_failed], to: :registered
+      transitions from: %i[no_account], to: :registered
     end
 
     event :upgrade do
-      transitions from: %i[unknown registered upgrade_failed], to: :upgraded
+      transitions from: %i[registered existing], to: :upgraded
     end
 
     event :fail_register do
-      transitions from: [:unknown], to: :register_failed
+      transitions from: [:no_account], to: :register_failed
     end
 
     event :fail_upgrade do
-      transitions from: %i[unknown registered], to: :upgrade_failed
+      transitions from: %i[registered existing], to: :upgrade_failed
     end
   end
+  # rubocop:enable Metrics/BlockLength
 
-  def eligible?
-    user.authorize :mhv_account_creation, :access?
+  def creatable?
+    may_register?
   end
 
-  def accessible?
-    return false if mhv_correlation_id.blank?
-    (user.loa3? || user.authn_context.include?('myhealthevet')) && (upgraded? || existing?)
+  def upgradable?
+    may_upgrade? && [nil, 'Basic', 'Advanced'].include?(account_level)
```

Comment:
this reads pretty well but could be a good use of `in?` ... `account_level.in?([nil, 'Basic', 'Advanced'])` or `account_level.in?(UPGRADABLE_LEVELS)`

Reply:
Upon second glance, I'm not sure we need the account level check. Seems like that would already be covered by the `unless` clause of the `upgrade` transition.

Reply:
I think it's better to be explicit in case MHV introduces new types.